File: solution_set1/2251. Number of Flowers in Full Bloom.py
##ss
import bisect
class Solution:
    def fullBloomFlowers(self, flowers: List[List[int]], persons: List[int]) -> List[int]:
        
        flowers1 = sorted([x[0] for x in flowers])
        flowers2 = sorted([x[1] for x in flowers])
        
        return [bisect.bisect_right(flowers1,x) - bisect.bisect_left(flowers2,x) for x in persons]
        
        
        

File: cpp_solutions/2251. Number of Flowers in Full Bloom.cpp
class Solution {
public:
    vector<int> fullBloomFlowers(vector<vector<int>>& flowers, vector<int>& persons) {
        
        vector<int> fstart,fend;
        vector<int> ans;
        for(auto& it: flowers){
            fstart.push_back(it[0]);
            fend.push_back(it[1]);
        }
        
        sort(fstart.begin(),fstart.end());
        sort(fend.begin(),fend.end());
        
        for(int it: persons){
            int starttime = upper_bound(fstart.begin(),fstart.end(),it) - fstart.begin(); //bisect_right
            int endtime = lower_bound(fend.begin(),fend.end(),it) - fend.begin(); // bisect_left
            ans.push_back(starttime - endtime);
            
            
        }
        return ans;
        
    }
};
